### PR TITLE
Service 498

### PR DESCRIPTION
**Pull request by lesjas**

### CODE REVIEW

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We could not parse the GitHub identity of the following contributors: **Leslie J**.
This is most likely caused by a git client misconfiguration; please make sure to:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails

File: src/cs-client/OpenFin.FDC3.Client/Connection.cs [hunk 1]
```diff
@@ -1,104 +1,111 @@
...
 {
     internal static class Connection
     {
-        private static Fin.ChannelClient channelClient;
+        internal static Action<Exception> ConnectionInitializationComplete;
+        private static ChannelClient channelClient;
+        private static Action<ChannelChangedPayload> channelChangedHandlers { get; set; }
         private static Action<ContextBase> contextListeners { get; set; }
-        private static Action<ChannelChangedEvent> channelChangedHandlers { get; set; }
         private static List<KeyValuePair<string, Action<ContextBase>>> intentListeners { get; set; }
```
Comment (nicholasdgoodman) [inline src/cs-client/OpenFin.FDC3.Client/Connection.cs]: Remove property accessors.
Consider `Dictionary<string,Action<ContextBase>>`

File: src/cs-client/OpenFin.FDC3.Client/DesktopAgent.cs [hunk 2]
```diff
@@ -43,13 +121,13 @@ public static void Initialize(Uri manifestUri)
                         fdcService.run();
                     }
 
-                    Connection.ConnectionInitializationComplete += () =>
+                    Connection.ConnectionInitializationComplete += exception =>
```
Comment (nicholasdgoodman) [inline src/cs-client/OpenFin.FDC3.Client/DesktopAgent.cs]: Does this never need to be unsubscribed?

**Review by nicholasdgoodman — CHANGES_REQUESTED** [review]
Only request is the property accessor removals. Otherwise at your discretion.

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We could not parse the GitHub identity of the following contributors: **Leslie J**.
This is most likely caused by a git client misconfiguration; please make sure to:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We could not parse the GitHub identity of the following contributors: **Leslie J**.
This is most likely caused by a git client misconfiguration; please make sure to:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We could not parse the GitHub identity of the following contributors: **Leslie J**.
This is most likely caused by a git client misconfiguration; please make sure to:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails

**Pull request merged**